subprojects/gradle/build-metrics/src/test/kotlin/com/avito/android/plugin/build_metrics/BuildMetricTrackerTest.kt

```diff
@@ -0,0 +1,84 @@
+package com.avito.android.plugin.build_metrics
+
+import com.avito.android.plugin.build_metrics.internal.BuildStatus
+import com.avito.android.sentry.StubEnvironmentInfo
+import com.avito.android.stats.SeriesName
+import com.avito.android.stats.StubStatsdSender
+import com.avito.android.stats.TimeMetric
+import com.avito.utils.gradle.Environment
+import com.google.common.truth.Truth.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+internal class BuildMetricTrackerTest {
+
+    private lateinit var statsd: StubStatsdSender
```

Comment:
```suggestion
    private val statsd = StubStatsdSender()
```

`lateinit` only adds complexity here; junit default blocking runner creates new instance on `BuildMetricTrackerTest` for each test invocation

![image](https://user-images.githubusercontent.com/1105133/120303896-f343dc00-c2d7-11eb-9e40-fb15813576c8.png)

Reply:
It has state. I want to recreate it explicitly. It seems easier for me, don't rely on implicit contract from test runner.

Reply:
i see, i prefer private factory function then, also without `lateinit`

Reply:
Fixed, changed to a factory and create in each test explicitly.